docs/conf.py

```diff
@@ -4,17 +4,40 @@
 # list see the documentation:
 # https://www.sphinx-doc.org/en/master/usage/configuration.html
 
+from functools import partial
+from pathlib import Path
+
+from setuptools_scm import get_version
```

Comment:
I note that `setuptools_scm` says "It is discouraged to use setuptools_scm from Sphinx itself": https://github.com/pypa/setuptools_scm/#usage-from-sphinx
thoughts?

Reply:
Sounds reasonable. Although, this code is what I use right now in many places and it works. Reading the version from metadata means that the project needs to be `pip install`ed into the same venv where sphinx is. It would subsequently pull more dependencies which is not possible or desirable in some cases.
OTOH it's not possible to switch the solution _right now_ because MyST does not use `setuptools-scm` to generate the distribution version — we'd not have the desired information before such a switch.
But if you want, I can contribute that separately, after this PR is in, for example.